File: src/main/java/com/example/message/mapper/ConversationMapper.java
package com.example.message.mapper;

import com.example.message.domain.Conversation;
import com.example.message.domain.dto.ConversationDto;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
public class ConversationMapper implements IMapper<Conversation, ConversationDto> {


    @Override
    public Conversation mapToDomain(ConversationDto conversationDto) {
        return new Conversation(conversationDto.getId(), userMapper.mapToDomainList(conversationDto.getWith()), conversationDto.isUnread());
    }

    @Override
    public ConversationDto mapToDto(Conversation conversation) {
        return new ConversationDto(conversation.getId(), userMapper.mapToDtoList(conversation.getWith()), conversation.isUnread());
    }

    @Autowired
    private UserMapper userMapper;
}

File: src/main/java/com/example/message/mapper/ConversationMapper.kt
package com.example.message.mapper

import com.example.message.domain.Conversation
import com.example.message.domain.dto.ConversationDto
import org.springframework.stereotype.Component

@Component
class ConversationMapper(private val userMapper: UserMapper) :
        IMapper<Conversation, ConversationDto> {

    override fun mapToDomain(dto: ConversationDto): Conversation {
        return Conversation(dto.id, userMapper.mapToDomainList(dto.with), dto.unread)
    }

    override fun mapToDto(domain: Conversation): ConversationDto {
        return ConversationDto(domain.id, userMapper.mapToDtoList(domain.with), domain.unread)
    }
}